tests/spark_test.py

```diff
@@ -0,0 +1,453 @@
+"""PySpark support in Narwhals is still _very_ limited.
+Start with a simple test file whilst we develop the basics.
+Once we're a bit further along, we can integrate PySpark tests into the main test suite.
+"""
+
+from __future__ import annotations
+
+from contextlib import nullcontext as does_not_raise
+from typing import TYPE_CHECKING
+from typing import Any
+
+import pandas as pd
+import pytest
+
+import narwhals.stable.v1 as nw
+from narwhals._exceptions import ColumnNotFoundError
+from tests.utils import NUMPY_VERSION
+from tests.utils import PYSPARK_VERSION
+from tests.utils import assert_equal_data
+
+if TYPE_CHECKING:
+    from pyspark.sql import SparkSession
+
+    from narwhals.typing import IntoFrame
+    from tests.utils import Constructor
+
+
+def _pyspark_constructor_with_session(obj: Any, spark_session: SparkSession) -> IntoFrame:
+    # NaN and NULL are not the same in PySpark
+    pd_df = pd.DataFrame(obj).replace({float("nan"): None}).reset_index()
+    return (  # type: ignore[no-any-return]
+        spark_session.createDataFrame(pd_df).repartition(2).orderBy("index").drop("index")
+    )
+
+
+@pytest.fixture(params=[_pyspark_constructor_with_session])
+def pyspark_constructor(
```

Comment:
does it work to add this to the `constructors` fixture and xfail whatever isn't supported yet? then we can see how much further we have to go

Reply:
ah I see, thanks

sure, we could switch over as a follow-up?